### PR TITLE
stdlib: Add |> and |< operators for result pipelines

### PR DESCRIPTION
**Pull request by V-FEXrt**
Result pipelines allow for much nicer user APIs (when designed for). This PR adds operators to allow for such pipelines

### CODE REVIEW

File: tests/wake-format/basic/stdout [hunk 1]
```diff
@@ -1783,3 +1783,20 @@ def a =
...
+    |< b
+    |< c
+
+def a =
+    a |> b
+    |< c
+
+def a =
+    a |< b
+    |> c
```
Comment (V-FEXrt) [inline tests/wake-format/basic/stdout:1802]: need to fix the formatting here

File: share/wake/lib/core/result.wake [hunk 2]
```diff
@@ -239,3 +239,25 @@ export def addErrorContext (prefix: String): Result a Error => Result a Error =
 #   else failWithError "Could not parse {str} as an Integer"
 export def failWithError (cause: String): Result a Error =
     Fail (makeError cause)
+
+# |>: Result compatible pipeline function. Used in the same way as '|'.
```
Comment (ag-eitilt) [inline share/wake/lib/core/result.wake:243]: Maybe?
```suggestion
# |>: Result compatible pipeline function. Used in the same way as '|' when both the input and the following function return a Result.
```

File: tests/wake-format/basic/stdout (hunk 1 above)
Comment (ag-eitilt) [reply tests/wake-format/basic/stdout:1802]: It's an interesting formatting bug, definitely!  Nice to have it exercising whatever odd edge case is in the code here.

**Review by ag-eitilt — APPROVED** [review]
Yeah, I like how this looks.

File: tests/wake-format/basic/stdout (hunk 1 above)
Comment (V-FEXrt) [reply tests/wake-format/basic/stdout]: Fixed!

File: tools/wake-format/emitter.cpp [hunk 3]
```diff
@@ -409,9 +431,25 @@ static std::vector<CSTElement> collect_right_binary(CSTElement collect_over, CST
...
+  // Ex: OP_OR("|") != OP_OR("||")
+  // Ex: OP_OR("||") == OP_OR("||")
+  bool matches = false;
+  for (const auto& i : collect_over) {
+    if (op.firstChildElement().id() == i.first &&
+        op.firstChildElement().fragment().segment().str() == i.second) {
+      matches = true;
+      break;
+    }
+  }
```
Comment (ag-eitilt) [inline tools/wake-format/emitter.cpp:450]: Is this significant enough to DRY out?  I don't see any differences from the other block at a glance.

File: tools/wake-format/emitter.cpp [hunk 4]
```diff
@@ -1361,9 +1399,14 @@ wcl::doc Emitter::walk_binary(ctx_t ctx, CSTElement node) {
 
   std::vector<CSTElement> parts = {};
   if (is_op_left_assoc(op_token)) {
-    parts = collect_left_binary(op_token, node);
+    if (is_binop_matching_str(op_token, TOKEN_OP_OR, "|>") ||
+        is_binop_matching_str(op_token, TOKEN_OP_OR, "|<")) {
+      parts = collect_left_binary({{TOKEN_OP_OR, "|>"}, {TOKEN_OP_OR, "|<"}}, node);
```
Comment (ag-eitilt) [inline tools/wake-format/emitter.cpp:1404]: This is the sort of thing we'll need to fix `,` and `++`, right?

**Review by ag-eitilt — APPROVED** [review]
Still looks good!

File: tools/wake-format/emitter.cpp (hunk 3 above)
Comment (V-FEXrt) [reply tools/wake-format/emitter.cpp:450]: Yeah its the same, I just didn't have a good enough name for it

File: tools/wake-format/emitter.cpp (hunk 4 above)
Comment (V-FEXrt) [reply tools/wake-format/emitter.cpp:1404]: I think so, not 100% sure. I think it might be bit more subtle. Its at least worth trying
